### ISSUE TITLE
[Feature][New resource] WAF (OCI WAAS) edge subnets q cartesian product of CIDRs x ports

### ISSUE THREAD

**Issue by denismakogon** [oracle/terraform-provider-oci#1461]
We'd like to see a more advanced WAF edge subnets NSG configuration.

This request is based on the technical limitation of the Terraform itself - it doesn't like it if the count value cannot be predetermined.
The issue we've faced recently:
```
╷
│ Error: Invalid for_each argument
│ 
│   on .terraform/modules/oke-v4/modules/network/nsgs.tf line 468, in resource "oci_core_network_security_group_security_rule" "waf_ingress":
│  468:   for_each                  = var.enable_waf == true ? toset(local.waf_cidr_list) : toset([])
│     ├────────────────
│     │ local.waf_cidr_list will be known only after apply
│     │ var.enable_waf is true
│ 
│ The "for_each" value depends on resource attributes that cannot be determined until apply, so Terraform cannot predict how many instances will be created. To work around this, use the -target argument to first apply only the
│ resources that the for_each depends on.
```

[Local var](https://github.com/oracle-terraform-modules/terraform-oci-oke/blob/4.x/modules/network/locals.tf#L48) that causes TF apply to fail:
```terraform
  waf_cidr_list = var.enable_waf == true ? [
    for waf_subnet in data.oci_waas_edge_subnets.waf_cidr_blocks[0].edge_subnets :
    waf_subnet.cidr
  ] : []
```

So, this type of technical limitation makes us run the TF module in two steps:
1. have a WAF NSG disabled, run apply.
2. enable WAF NSG, run apply one more time.

What we'd like to see is a new type of resource that can do the cartesian product of CIDRs x ports.

Parameters to a new resource:
- list of CIDRs
- list of ports or port ranges

Attributes:
- TBA

**Comment by ravinitp** [comment]
We are very sorry that we couldn't respond to each and every issue reported on GitHub. 
Although we have refined the process to prioritize customer issues on GitHub, since this issue was reported a while ago, there is a good chance it may have been fixed in the latest version of Terraform Provider OCI. 
If you are still experiencing this issue, please create a new issue and label it as Bug.
